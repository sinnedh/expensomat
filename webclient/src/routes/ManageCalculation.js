import React from 'react'
import { connect } from 'react-redux'
import MembersList from '../components/MembersList'
import { setToken } from '../actions'

class ManageCalculation extends React.Component {
  componentDidMount() {
    this.props.onComponentDidMount()
  }

  render() {
    return (
      <div>
        <h1>Manage calculation: {this.props.name}</h1>
        <span>{this.props.description}</span>
        <h2>Members</h2>
        <MembersList members={this.props.members} />
      </div>
    )
  }
}

const mapStateToProps = (state, ownProps) => ({
  name: state.calculations.name,
  description: state.calculations.description,
  members: state.calculations.members,
})

const mapDispatchToProps = (dispatch, ownProps) => ({
  onComponentDidMount: () => {
    const token = ownProps.match.params.token
    dispatch(setToken(token))
  },
})

export default connect(mapStateToProps, mapDispatchToProps)(ManageCalculation)
