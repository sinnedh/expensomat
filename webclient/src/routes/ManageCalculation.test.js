import React from 'react'
import { List, Map } from 'immutable'
import configureStore from 'redux-mock-store'
import ManageCalculation from './ManageCalculation'

describe('ManageCalculation', () => {
  it('renders properly', () => {
    const initialState = Map({
      application: Map({ token: 'ABCD1234' }),
      calculations: Map({
        name: 'My calculation',
        description: 'Blabla',
      }),
      members: Map({
        1: Map({id: 1, name: 'A guy'}),
        2: Map({id: 2, name: 'Another guy'}),
      }),
    })
    const mockStore = configureStore()
    const store = mockStore(initialState)
    const wrapper = shallow(
      <ManageCalculation
        match={{params: {token: 'ABCD1234'}}}
        store={store}
        />
    )
    expect(wrapper.dive()).toMatchSnapshot()
  })
})
