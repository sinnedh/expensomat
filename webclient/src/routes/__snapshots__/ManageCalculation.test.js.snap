// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManageCalculation renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ManageCalculation
    description="Blabla"
    match={
        Object {
            "params": Object {
              "token": "ABCD1234",
            },
          }
    }
    members={
        Object {
            "1": Object {
              "id": 1,
              "name": "A guy",
            },
            "2": Object {
              "id": 2,
              "name": "Another guy",
            },
          }
    }
    name="My calculation"
    onUpdateDescription={[Function]}
    onUpdateMemberName={[Function]}
    onUpdateName={[Function]}
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
    storeSubscription={
        Subscription {
            "listeners": Object {
              "clear": [Function],
              "get": [Function],
              "notify": [Function],
              "subscribe": [Function],
            },
            "onStateChange": [Function],
            "parentSub": undefined,
            "store": Object {
              "clearActions": [Function],
              "dispatch": [Function],
              "getActions": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            },
            "unsubscribe": [Function],
          }
    }
    token="ABCD1234"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Manage calculation:
</h1>,
        <div>
          <label>
                    Name:
          </label>
           
          <EditableInput
                    onClickSave={[Function]}
                    value="My calculation"
          />
</div>,
        <div>
          <label>
                    Description:
          </label>
           
          <EditableInput
                    onClickSave={[Function]}
                    value="Blabla"
          />
</div>,
        <h2>
          Members
</h2>,
        <ul>
          <li>
                    <EditableInput
                              onClickSave={[Function]}
                              value="A guy"
                    />
          </li>
          <li>
                    <EditableInput
                              onClickSave={[Function]}
                              value="Another guy"
                    />
          </li>
</ul>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Manage calculation:",
        },
        "ref": null,
        "rendered": "Manage calculation:",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label>
              Name:
</label>,
            " ",
            <EditableInput
              onClickSave={[Function]}
              value="My calculation"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Name:",
            },
            "ref": null,
            "rendered": "Name:",
            "type": "label",
          },
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onClickSave": [Function],
              "value": "My calculation",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label>
              Description:
</label>,
            " ",
            <EditableInput
              onClickSave={[Function]}
              value="Blabla"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Description:",
            },
            "ref": null,
            "rendered": "Description:",
            "type": "label",
          },
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onClickSave": [Function],
              "value": "Blabla",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Members",
        },
        "ref": null,
        "rendered": "Members",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <EditableInput
                            onClickSave={[Function]}
                            value="A guy"
              />
</li>,
            <li>
              <EditableInput
                            onClickSave={[Function]}
                            value="Another guy"
              />
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": <EditableInput
                onClickSave={[Function]}
                value="A guy"
/>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "onClickSave": [Function],
                "value": "A guy",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": <EditableInput
                onClickSave={[Function]}
                value="Another guy"
/>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "onClickSave": [Function],
                "value": "Another guy",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Manage calculation:
</h1>,
          <div>
            <label>
                        Name:
            </label>
             
            <EditableInput
                        onClickSave={[Function]}
                        value="My calculation"
            />
</div>,
          <div>
            <label>
                        Description:
            </label>
             
            <EditableInput
                        onClickSave={[Function]}
                        value="Blabla"
            />
</div>,
          <h2>
            Members
</h2>,
          <ul>
            <li>
                        <EditableInput
                                    onClickSave={[Function]}
                                    value="A guy"
                        />
            </li>
            <li>
                        <EditableInput
                                    onClickSave={[Function]}
                                    value="Another guy"
                        />
            </li>
</ul>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Manage calculation:",
          },
          "ref": null,
          "rendered": "Manage calculation:",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label>
                Name:
</label>,
              " ",
              <EditableInput
                onClickSave={[Function]}
                value="My calculation"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Name:",
              },
              "ref": null,
              "rendered": "Name:",
              "type": "label",
            },
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "onClickSave": [Function],
                "value": "My calculation",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label>
                Description:
</label>,
              " ",
              <EditableInput
                onClickSave={[Function]}
                value="Blabla"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Description:",
              },
              "ref": null,
              "rendered": "Description:",
              "type": "label",
            },
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "onClickSave": [Function],
                "value": "Blabla",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Members",
          },
          "ref": null,
          "rendered": "Members",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <EditableInput
                                onClickSave={[Function]}
                                value="A guy"
                />
</li>,
              <li>
                <EditableInput
                                onClickSave={[Function]}
                                value="Another guy"
                />
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": <EditableInput
                  onClickSave={[Function]}
                  value="A guy"
/>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "onClickSave": [Function],
                  "value": "A guy",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "children": <EditableInput
                  onClickSave={[Function]}
                  value="Another guy"
/>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "onClickSave": [Function],
                  "value": "Another guy",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
